src/crate/client/connection.py

```diff
@@ -197,14 +199,21 @@ def get_blob_container(self, container_name):
 
     def _lowest_server_version(self):
         lowest = None
+        server_count = len(self.client.active_servers)
+        connection_errors = []
         for server in self.client.active_servers:
             try:
                 _, _, version = self.client.server_infos(server)
                 version = Version(version)
-            except (ValueError, ConnectionError):
+            except ConnectionError as ex:
+                connection_errors.append(ex)
+                continue
+            except (ValueError, InvalidVersion):
                 continue
             if not lowest or version < lowest:
                 lowest = version
+        if connection_errors and len(connection_errors) == server_count:
+            raise ConnectionError(json.dumps(list(map(str, connection_errors))))
```

Comment:
why not just `raise ConnectionError(error_list)`?

Reply:
I've elaborated about it [here](https://github.com/crate/crate-python/pull/711#discussion_r2081480875), but I am also not sure, that's why I am also asking about your opinion.